apps/bmd/src/AppRoot.tsx

```diff
@@ -141,898 +147,1091 @@ export const activationStorageKey = 'activation'
-        const isRecentVoterPrint =
-          isVoterCardPrinted &&
-          utcTimestamp() <=
-            ballotPrintedTime + GLOBALS.RECENT_PRINT_EXPIRATION_SECONDS
-        const ballotStyle = getBallotStyle({
-          election: election!,
-          ballotStyleId: voterCardData.bs,
-        })
-        const precinct = getPrecinctById({
-          election: election!,
-          precinctId: voterCardData.pr,
-        })
-        const isVoterCardValid = Boolean(ballotStyle) && Boolean(precinct)
-
-        const ballot: Partial<CompletedBallot> =
-          (longValueExists &&
-            this.state.isVoterCardValid &&
-            !this.state.isVoterCardExpired &&
-            !this.state.isVoterCardVoided &&
-            !this.state.isVoterCardPrinted &&
-            (await this.fetchBallotData())) ||
-          {}
-
-        this.setState((prevState) => {
-          const isVoterCardExpired =
-            prevState.voterCardCreatedAt === 0 &&
-            utcTimestamp() >=
-              voterCardCreatedAt + GLOBALS.CARD_EXPIRATION_SECONDS
-
-          return {
-            ...this.initialCardPresentState,
-            shortValue,
-            isVoterCardExpired,
-            isVoterCardVoided,
-            isVoterCardPresent: true,
-            isVoterCardPrinted,
-            isRecentVoterPrint,
-            isVoterCardValid,
-            voterCardCreatedAt,
-            ballotStyleId:
-              ballotStyle?.id ?? this.initialAppState.ballotStyleId,
-            precinctId: precinct?.id ?? this.initialAppState.precinctId,
-            votes: ballot.votes,
-            contests:
-              ballotStyle && election
-                ? getContests({ ballotStyle, election })
-                : this.initialAppState.contests,
+          break
+        }
+        case 'pollworker': {
+          dispatchState({ type: 'processPollWorkerCard' })
+          break
+        }
+        case 'admin': {
+          /* istanbul ignore else */
+          if (longValueExists) {
+            dispatchState({ type: 'processAdminCard' })
           }
-        })
-        if (precinct && ballotStyle) {
-          this.storeBallotActivation({
-            ballotCreatedAt: voterCardData.c,
-            ballotStyleId: ballotStyle.id,
-            precinctId: precinct.id,
-          })
+          break
         }
-
-        break
-      }
-      case 'pollworker': {
-        this.setState({
-          ...this.initialCardPresentState,
-          isPollWorkerCardPresent: true,
-        })
-        break
       }
-      case 'admin': {
-        /* istanbul ignore else */
-        if (longValueExists) {
-          this.setState({
-            ...this.initialCardPresentState,
-            isAdminCardPresent: true,
-          })
+    },
+    [card, optionalElectionDefinition]
+  )
+
+  const [
+    startCardShortValueReadPolling,
+    stopCardShortValueReadPolling,
+  ] = useCallback(
+    useInterval(async () => {
```

Comment:
This is a pretty strange construct. `useCallback` typically takes a function as its first argument, not a tuple/array. Also, we're using v3.6.0 of `@rooks/use-interval`, which is not the latest, and it has broken TypeScript typings which leads TS to consider the types of `startCardShortValueReadPolling` etc to be `any`. Is there a reason we can't use v4.5.0?

Reply:
I'm also confused what it means to have useInterval inside of a useCallback here, why are we even wrapping this in useCallback?

Reply:
@eventualbuddha FTW! https://github.com/votingworks/vxsuite/pull/126/commits/b0901479d73d68c12be4727c6b3c7e1a507d5bfa